etc/tests/test_bootstrapper.py

```diff
@@ -163,6 +164,183 @@ def test_main_method_with_dir_outputs(monkeypatch, s3_setup, tmpdir):
+
+    Verifies that process_metrics produces a valid KFP UI metadata file if
+    the node's script | notebook did not generate this metadata file.
+    """
+    argument_dict = {'cos-endpoint': 'http://' + MINIO_HOST_PORT,
+                     'cos-bucket': 'test-bucket',
+                     'cos-directory': 'test-directory',
+                     'cos-dependencies-archive': 'test-archive.tgz',
+                     'filepath': 'etc/tests/resources/test-notebookA.ipynb',
+                     'inputs': 'test-file.txt;test,file.txt',
+                     'outputs': 'test-file/test-file-copy.txt;test-file/test,file/test,file-copy.txt',
+                     'user-volume-path': None}
+
+    metadata_file = '/tmp/mlpipeline-ui-metadata.json'
+
+    os.remove(metadata_file)
+
+    main_method_setup_execution(monkeypatch, s3_setup, tmpdir, argument_dict)
+    # process_metrics generates a file named
+    # /tmp/mlpipeline-ui-metadata.json should now be present
+
+    try:
+        with open(metadata_file, 'r') as f:
+            metadata = json.load(f)
+            assert metadata.get('outputs') is not None
+            assert isinstance(metadata['outputs'], list)
+            assert len(metadata['outputs']) == 1
+            assert metadata['outputs'][0]['storage'] == 'inline'
+            assert metadata['outputs'][0]['type'] == 'markdown'
+            assert '{}/{}/{}'.format(argument_dict['cos-endpoint'],
+                                     argument_dict['cos-bucket'],
+                                     argument_dict['cos-directory']) \
+                in metadata['outputs'][0]['source']
+            assert argument_dict['cos-dependencies-archive']\
+                in metadata['outputs'][0]['source']
+    except AssertionError:
+        raise
+    except Exception as ex:
+        # Potential reasons for failures:
+        # file not found, invalid JSON
+        print('Validation of "{}" failed: {}'.format(str(ex), ex))
+        assert False
+
+
+def test_process_metrics_method_valid_metadata_file(monkeypatch, s3_setup, tmpdir):
+    """Test for process_metrics
+
+    Verifies that process_metrics produces a valid KFP UI metadata file if
+    the node's script | notebook did already generate this metadata file. The
+    content of that file should be preserved.
+    """
+    argument_dict = {'cos-endpoint': 'http://' + MINIO_HOST_PORT,
+                     'cos-bucket': 'test-bucket',
+                     'cos-directory': 'test-directory',
+                     'cos-dependencies-archive': 'test-archive.tgz',
+                     'filepath': 'etc/tests/resources/test-notebookA.ipynb',
+                     'inputs': 'test-file.txt;test,file.txt',
+                     'outputs': 'test-file/test-file-copy.txt;test-file/test,file/test,file-copy.txt',
+                     'user-volume-path': None}
+
+    input_metadata_file = 'mlpipeline-ui-metadata.json'
+    output_metadata_file = '/tmp/mlpipeline-ui-metadata.json'
```

Comment:
ditto

Reply:
The test will now be skipped if the environment doesn't meet the file system prerequisites.